visits.md

```diff
@@ -0,0 +1,249 @@
+# Visits 
+
+### Overview
+
+* A Visit in OpenMRS represents precisely what it sounds like: a time when a patient is actively interacting with the 
+the healthcare system, typically at a location.
+
+* The metadata differentiating different types of visits is a Visit Type. Visit Types displayed in the user interface, 
+also, can be searched against.
+
+* A visit contains encounters, which store more granular data about treatments or services.
+
+##### Let’s look at an example.
+
+At the Amani Clinic, a patient might typically check-in at registration, be seen by a doctor, and receives medication 
+<b> dispensed </b> in the pharmacy. This would be recorded as one <b> visit </b> of <b> visit type of Outpatient </b>, and 
+contain <b> three encounters (Registration, Consultation, and Dispensing) </b>.
+
+### Sub Resource types
+
+#### Attribute.
+
+* If you wish to record extra information about visits, you can create Visit Attributes and assign them to Visit Types.
+
+* Visit attributes exists specifically to allow implementations to extend the data model.
+
+
+### Available operations. 
+<ol>
+    <li>List visits.</li>
+    <li>Create a visit.</li>
+    <li>Update a visit.</li>
+    <li>Delete a visit.</li>
+    <li>List attribute sub resources.</li>
+    <li>Create attribute sub resource with properties.</li>
+    <li>Update attribute sub resource.</li>
+    <li>Delete attribute sub resource.</li>
+</ol>
+
+
+#### List visits.
+
+* **List all non-retired visits.**
+    
+    Quickly filter visits with given query parameters.Returns a `404 Not Found` status if visit not exists. If user not logged 
+    in to perform this action,a `401 Unauthorized` status returned.
+    
+    ##### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *patient* | `String` | **Patient resource UUID**
+    *location* | `String` | **Location resource UUID**
+    *includeInactive* | `Boolean` | **Active/Inactive status of visit**
+    *fromStartDate* | `Date (ISO8601 Long)` | **Start date of the visit**
+
+    ```console
+    curl -X GET 'https://qa-refapp.openmrs.org/openmrs/ws/rest/v1/visit?
+      includeInactive=true
+      &fromStartDate=2016-10-08T04:09:23.000Z
+      &patient=target_patient_uuid
+      &location=target_location_uuid'
+    -H 'Authorization: Basic Auth'
+     ```
+    
+* **List visit by UUID.**
+
+    Retrieve a visit by its UUID. Returns a `404 Not Found` status if visit not exists. If user not logged 
+    in to perform this action, a `401 Unauthorized` status returned.
+    
+    ```console
+    curl -X GET "/openmrs/ws/rest/v1/visit/target_visit_uuid" 
```

Comment:
Use the convention of prefixing variables with a colon (`:`). For example:

`GET /visit/:target_visit_uuid`

Reply:
done